### PR TITLE
fix: improve home semantics

### PR DESCRIPTION
**Pull request by greatislander**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

Uses labelled `<section>` elements for improved home page semantics; removes `<hr />` elements in favour of pure CSS solution for horizontal dividers.

Also: splits typography styles into partials.

## Steps to test

Review layout: https://deploy-preview-179--pinecone.netlify.com/components/preview/home.html

## Additional information

Resolves changes requested in https://github.com/platform-coop-toolkit/coop-library/pull/151#pullrequestreview-346175701.

## Related issues

Not applicable.

### CODE REVIEW

File: src/components/03-layouts/04-home/home.njk [hunk 1]
```diff
@@ -57,7 +57,6 @@
 			{% render '@info-card', {standAlone: true, icon: false, title: 'Not sure how to use the library?', description: false, actionLabel: 'Learn how'}, true %}
 		</div>
```
Comment (jhung) [inline src/components/03-layouts/04-home/home.njk]: The info-cards container can be put into an <aside> element.

Comment (jhung) [inline src/components/03-layouts/04-home/home.njk]: The info-cards `<div>` can be an `<aside>` element.

File: src/components/03-layouts/04-home/home.njk [hunk 2]
```diff
@@ -2,11 +2,12 @@
 	<h1>{{ title | safe }}</h1>
 	<p class="subhead">{{ subhead }}</p>
 </div>
-<div class="home__search">
+<section class="home__search">
+	<h2 class="screen-reader-text">{{ search.title }}</h2>
```
Comment (jhung) [inline src/components/03-layouts/04-home/home.njk]: After reviewing this with NVDA, this `<h2>Search</h2>` may not be necessary. The Search landmark already gets announced.

**Comment by greatislander** [comment]
@jhung Fixed menu component.

**Pull request merged**